db/versions/0023.yml

```diff
@@ -0,0 +1,241 @@
+      end
+  get_index_namespace:
+    description: |-
+      Get a namespace. The returned table will have one or zero rows.
+    mode: read
+    serviceName: index
+    args: parent_in text, name_in text
+    returns: table(parent text, name text, expires timestamptz)
+    body: |-
+      begin
+        return query select
+          index_namespaces.parent,
+          index_namespaces.name,
+          index_namespaces.expires
+        from index_namespaces
+        where
+          index_namespaces.parent = parent_in and
+          index_namespaces.name = name_in;
+      end
+  update_index_namespace:
+    serviceName: index
+    description: |-
+      Update a namespace.
+      Returns the up-to-date namespace row that have the same parent and name.
+      If the row is not found then an exception with code 'P0002' is thrown.
+    mode: write
+    args: parent_in text, name_in text, expires_in timestamptz
+    returns: table(parent text, name text, expires timestamptz)
+    body: |-
+      declare
+        updated_row index_namespaces%ROWTYPE;
+      begin
+        update index_namespaces
+        set expires = coalesce(expires_in, index_namespaces.expires)
+        where
+          index_namespaces.parent = parent_in and
+          index_namespaces.name = name_in
+        returning
+          index_namespaces.parent,
+          index_namespaces.name,
+          index_namespaces.expires
+        into updated_row;
+        if found then
+          return query select
+            updated_row.parent,
+            updated_row.name,
+            updated_row.expires;
+          return;
+        end if;
+        raise exception 'no such row' using errcode = 'P0002';
+      end
+  expire_index_namespaces:
+    description: |-
+      Expire index_namespaces that come before the current time.
+      Returns a count of rows that have been deleted.
+    mode: write
+    serviceName: index
+    args: ''
+    returns: integer
+    body: |-
+      declare
+        count integer;
+      begin
+        delete from index_namespaces where index_namespaces.expires < now();
+        if found then
+          get diagnostics count = row_count;
+          return count;
+        end if;
+        return 0;
+      end
+  get_indexed_tasks:
+    description: |-
+      Get existing indexed tasks filtered by the optional arguments,
+      ordered by the `namespace` and `name`.
+      If the pagination arguments are both NULL, all rows are returned.
+      Otherwise, page_size rows are returned at offset page_offset.
+    mode: read
+    serviceName: index
+    args: namespace_in text, name_in text, page_size_in integer, page_offset_in integer
+    returns: table(namespace text, name text, rank integer, task_id text, data jsonb, expires timestamptz)
+    body: |-
+      begin
+        return query
+        select
+          indexed_tasks.namespace,
+          indexed_tasks.name,
+          indexed_tasks.rank,
+          indexed_tasks.task_id,
```

Comment:
Unfortunately, this is a uuid and not a slugid, so we'll need to call uuid_to_slugid here, and in general call that or slugid_to_uuid in all of these functions.  The functions should expose slugids as their interface, and then later we can figure out how to migrate the table to have slugids natively (maybe getting "creative" and doing so in a way with no downtime).